src/main/java/com/minecolonies/coremod/network/messages/BuildToolPlaceMessage.java

```diff
@@ -132,8 +132,9 @@ public void messageOnServerThread(final BuildToolPlaceMessage message, final Ent
      * @param rotation The number of times the schematic should be rotated.
      * @param buildPos The location the hut is being placed.
      */
-    private static void handleHut(@NotNull final World world, @NotNull final EntityPlayer player,
-            final String hut, final String style, final int rotation, @NotNull final BlockPos buildPos)
+    private static void handleHut(
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "handleHut" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)